src/main/java/it/gov/pagopa/reward/service/synchronous/RewardTrxSynchronousApiServiceImpl.java

```diff
@@ -0,0 +1,97 @@
+                .flatMap(b -> userInitiativeCountersRepository.findById(UserInitiativeCounters.buildId(trxDTO.getUserId(),initiativeId)))
+                .map(userInitiativeCounters -> {
+                    UserInitiativeCountersWrapper counterWrapper = UserInitiativeCountersWrapper.builder()
+                            .userId(userInitiativeCounters.getUserId())
+                            .initiatives(Map.of(initiativeId, userInitiativeCounters))
+                            .build();
+                    return initiativesEvaluatorFacadeService.evaluateInitiativesBudgetAndRules(trxDTO, List.of(initiativeId), counterWrapper);
+                })
+                .map(p -> rewardTransaction2SynchronousTransactionResponseDTOMapper.apply(trxPreviewRequest.getTransactionId(),initiativeId, p.getSecond()));
+    }
+
+    @Override
+    public Mono<SynchronousTransactionResponseDTO> authorizeTransaction(SynchronousTransactionRequestDTO trxAuthorizeRequest, String initiativeId) {
+        log.trace("[REWARD] Starting reward preview calculation for transaction {}", trxAuthorizeRequest.getTransactionId());
+        TransactionDTO trxDTO = trxPreviewRequest2TransactionDtoMapper.apply(trxAuthorizeRequest);
+        return checkInitiative(trxAuthorizeRequest, initiativeId)
+                .flatMap(b -> checkSynTrxAlreadyProcessed(trxAuthorizeRequest,trxDTO, initiativeId))
+                .flatMap(b -> checkOnboarded(trxAuthorizeRequest, trxDTO, initiativeId))
+                .flatMap(b -> userInitiativeCountersRepository.updateDate(trxAuthorizeRequest, initiativeId))
+                .flatMap(userInitiativeCounters -> initiativesEvaluatorFacadeService.evaluateAndUpdateBudget(trxDTO,List.of(initiativeId)))
+                .map(rewardTransaction -> rewardTransaction2SynchronousTransactionResponseDTOMapper.apply(trxAuthorizeRequest.getTransactionId(),initiativeId, rewardTransaction));
+
+
+    }
+
+    private Mono<Boolean> checkInitiative(SynchronousTransactionRequestDTO request, String initiativeId){
+        return rewardContextHolderService.getInitiativeConfig(initiativeId).hasElement()
+                .map(b -> checkingResult(b, request, initiativeId, RewardConstants.TRX_REJECTION_REASON_INITIATIVE_NOT_FOUND));
+    }
+    private Mono<Boolean> checkOnboarded(SynchronousTransactionRequestDTO request, TransactionDTO trx, String initiativeId){
+        return onboardedInitiativesService.isOnboarded(trx.getHpan(),trx.getTrxDate(), initiativeId)
+                .map(b -> checkingResult(b, request, initiativeId, RewardConstants.TRX_REJECTION_REASON_NO_INITIATIVE));
+    }
+
+    private Mono<Boolean> checkSynTrxAlreadyProcessed(SynchronousTransactionRequestDTO request,TransactionDTO trx, String initiativeId){
+        return transactionProcessedService.checkDuplicateTransactions(trx).hasElement()
+                .map(b -> checkingResult(b, request, initiativeId, RewardConstants.TRX_REJECTION_ALREADY_PROCESSED));
```

Comment:
here you should return 409 HttpStatus, the result returned is the transactionProcessed (mapped) itself, you have not to add rejectionReasons